### PR TITLE
Integrate buildconfig plugin to project to replace old version templates work

### PR DESCRIPTION
**Pull request by ZacSweers**
This uses a nicer plugin to handle buildconfig constants

### CODE REVIEW

File: slack-plugin/version-templates/KotlinBuildConfig.kt [hunk 1]
```diff
@@ -1,9 +0,0 @@
-@file:JvmName("KotlinBuildConfig")
```
Comment (linhpha) [inline slack-plugin/version-templates/KotlinBuildConfig.kt:1]: Why is this removed? Still seeing its reference here
https://github.com/slackhq/slack-gradle-plugin/pull/677/files#diff-c0dfa6bc7a8685217f70a860145fbdf416d449eaff052fa28352c5cec1a98c06R180.

Comment (ZacSweers) [reply slack-plugin/version-templates/KotlinBuildConfig.kt:1]: That's different than this file actually, they're just named the same. That other def is here: https://github.com/slackhq/slack-gradle-plugin/pull/677/files#diff-c0dfa6bc7a8685217f70a860145fbdf416d449eaff052fa28352c5cec1a98c06R122

File: build.gradle.kts [hunk 2]
```diff
@@ -186,16 +180,18 @@ val kotlinVersion = libs.versions.kotlin.get()
...
-    .matching { it.name == "copyVersionTemplates" }
-    .configureEach {
-      val templatesMap = kotlinBuildConfig.asTemplatesMap()
-      inputs.property("buildversions", templatesMap.hashCode())
-      expand(templatesMap)
+  if (project.path == ":slack-plugin") {
+    project.pluginManager.withPlugin("com.github.gmazzo.buildconfig") {
+      configure<BuildConfigExtension> {
+        buildConfigField("String", "KOTLIN_VERSION", "\"$kotlinVersion\"")
+        // Using Any here due to https://github.com/gmazzo/gradle-buildconfig-plugin/issues/9
```
Comment (ZacSweers) [inline build.gradle.kts:187]: PR'd a hopeful fix here https://github.com/gmazzo/gradle-buildconfig-plugin/pull/98
